### ISSUE TITLE
notes about 1.1.1_stable and DEVRANDOM_WAIT

### ISSUE THREAD

**Issue by petrovr** [openssl/openssl#9757]
**1)**  Android build with platform headers
```
arm-linux-androideabi-gcc  ... crypto/rand/rand_unix.c
crypto/rand/rand_unix.c:25:22: fatal error: sys/shm.h: No such file or directory
 # include <sys/shm.h>
                      ^
compilation terminated.
Makefile:4620: recipe for target 'crypto/rand/rand_unix.o' failed
```

**Solution**: include sys/shm.h only if DEVRANDOM_WAIT is defined

**2)** After correction of 1.)
```
arm-linux-androideabi-gcc ...  crypto/rand/rand_unix.c
crypto/rand/rand_unix.c: In function 'syscall_random':
crypto/rand/rand_unix.c:345:5: warning: implicit declaration of function 'syscall' [-Wimplicit-function-declaration]
     return syscall(__NR_getrandom, buf, buflen, 0);
```
Linux manual pages provides following information:
```
       #define _GNU_SOURCE         /* See feature_test_macros(7) */
       #include <unistd.h>
       #include <sys/syscall.h>   /* For SYS_xxx definitions */

       long syscall(long number, ...);
```
**Solution**: include <sys/syscall.h> if defined OPENSSL_SYS_LINUX

**3)** include header **sys/ipc.h**
It seems to me header is included only for mode_t, but mode_t is defined in sys/stat.h as well and sys/stat.h it is included below in statement if defined(OPENSSL_SYS_UNIX) 

**Solution**:  to remove

**4)** include header **sys/utsname.h**
It is used only if DEVRANDOM_WAIT is defined

**Solution**:  include only if DEVRANDOM_WAIT is defined

**5)** include header **asm/unistd.h**
It is included from sys/syscall.h, i..e see point **2)** above

**Solution**:  to remove

In summary: 
```diff
--- a/crypto/rand/rand_unix.c
+++ b/crypto/rand/rand_unix.c
@@ -19,12 +19,6 @@
 #include "internal/rand_int.h"
 #include <stdio.h>
 #include "internal/dso.h"
-#if defined(__linux)
-# include <asm/unistd.h>
-# include <sys/ipc.h>
-# include <sys/shm.h>
-# include <sys/utsname.h>
-#endif
 #if defined(__FreeBSD__) && !defined(OPENSSL_SYS_UEFI)
 # include <sys/types.h>
 # include <sys/sysctl.h>
@@ -40,6 +34,13 @@
 # include <fcntl.h>
 # include <unistd.h>
 # include <sys/time.h>
+# ifdef DEVRANDOM_WAIT
+#  include <sys/shm.h>
+#  include <sys/utsname.h>
+# endif
+# ifdef OPENSSL_SYS_LINUX
+#  include <sys/syscall.h>
+# endif
 
 static uint64_t get_time_stamp(void);
 static uint64_t get_timer_bits(void);
```

**Comment by petrovr** [comment]
**6)** if !defined(FIPS_MODE)
Once again inappropriate to 1.1.1 branch code.

Please  clean-up

7) defines of ```__NR_getrandom``` (informative)
Thanks.
Unlike unified headers platform headers lack those definitions. In unified ```__NR_ foo``` are defined unconditionally. 
FYI on "old" devices OS return ENOSYS.

8) shared memory (informative)
Android supports shared-memory. Device is /dev/ashmem .  Implementation differ from POSIX as is related to application. If I remember well if application exit shared memory is released.
With other words is not suitable for "global" flag.
